### DISCUSSION TITLE
How do I get Rector to bypass/skip problem files and list them out?

### DISCUSSION THREAD

**Discussion by 7thstorm** [rectorphp/rector#8011] [Q&A]
I'm spending a lot of time re-running the process because it finds a file it does not like and stops there. I rename it and it finds another one it does not like. 
Is there a flag or some option I can use to simply skip files that rector has problems with? Perhaps I can find the names in the log or perhaps it can spit out the names as exceptions.

Hope someone has a solution!

**Comment by TomasVotruba** [comment]
Hi, you can use the `--debug` option to get more info.

We should fix the buggy case instead. What exactly is your issue?

**Reply by 7thstorm** [reply]
Thanks.  I am using the following command:

(It's a windows 11 pro machine)

`php vendor\bin\rector process my-folder\my-path --dry-run -vvv --debug --output-format=json > rector_folder-path.txt`

My Rector.php file is this:

```
<?php

declare(strict_types=1);

use Rector\CodeQuality\Rector\Class_\InlineConstructorDefaultToPropertyRector;
use Rector\Config\RectorConfig;
//use Rector\Set\ValueObject\LevelSetList;
use Rector\Set\ValueObject\SetList; // added
use Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator; 

return static function (RectorConfig $rectorConfig): void {
    $rectorConfig->paths([
        __DIR__ . '/my-folder',
    ]);

    // Skip certain paths
    $rectorConfig->skip([
       
    ]);

    // configure parallel processing
    $rectorConfig->parallel(
        $seconds = 1200,
        $maxNumberOfProcess = 14,
        $jobSize = 20
    );

    // register a single rule
    //$rectorConfig->rule(InlineConstructorDefaultToPropertyRector::class);

    // define sets of rules
    $rectorConfig->sets([
		SetList::MYSQL_TO_MYSQLI,
    ]);
};
```

The output file does not show any errors.
Each time, it stops at some file and the last line in the output is:

```
[file] my-folder/my-subfolder/filename.php
[rule] Rector\MysqlToMysqli\Rector\FuncCall\MysqlQueryMysqlErrorWithLinkRector
```

The files are much longer than 100 lines so I'm not able to put them in the rector demo page.

are the detailed logs stored somewhere else?

**Reply by 7thstorm** [reply]
I have some additional information. I'm hoping someone can guide me.

When I use single rules (one at a time and all 4 together, the scan completes, and shows no changes, even though there are plenty of places it could change. BUT, the important thing is it at least completes the scan

here is the rector.php I'm using for single rules

```
<?php

declare(strict_types=1);

use Rector\CodeQuality\Rector\Class_\InlineConstructorDefaultToPropertyRector;
use Rector\Config\RectorConfig;
use Rector\Set\ValueObject\LevelSetList;
use Rector\Set\ValueObject\SetList;

return static function (RectorConfig $rectorConfig): void {
    $rectorConfig->paths([
        __DIR__ . '/code-files',
    ]);

    // register a single rule
    $rectorConfig->rule(\Rector\MysqlToMysqli\Rector\Assign\MysqlAssignToMysqliRector::class);
    $rectorConfig->rule(\Rector\MysqlToMysqli\Rector\FuncCall\MysqlFuncCallToMysqliRector::class);
    $rectorConfig->rule(\Rector\MysqlToMysqli\Rector\FuncCall\MysqlPConnectToMysqliConnectRector::class);
    $rectorConfig->rule(\Rector\MysqlToMysqli\Rector\FuncCall\MysqlQueryMysqlErrorWithLinkRector::class);


};
```

BUT, when I run these rules as a set, it hangs up at this rule:
`[rule] Rector\MysqlToMysqli\Rector\FuncCall\MysqlQueryMysqlErrorWithLinkRector`

**_This happens regardless of which file I try to scan_**

here is my rector.php for when using the set:

```
<?php

declare(strict_types=1);

use Rector\CodeQuality\Rector\Class_\InlineConstructorDefaultToPropertyRector;
use Rector\Config\RectorConfig;
use Rector\Set\ValueObject\LevelSetList;
use Rector\Set\ValueObject\SetList;

return static function (RectorConfig $rectorConfig): void {
    $rectorConfig->paths([
        __DIR__ . '/code-files',
    ]);



    // register a single rule
    //$rectorConfig->rule(\Rector\MysqlToMysqli\Rector\Assign\MysqlAssignToMysqliRector::class);
    //$rectorConfig->rule(\Rector\MysqlToMysqli\Rector\FuncCall\MysqlFuncCallToMysqliRector::class);
    //$rectorConfig->rule(\Rector\MysqlToMysqli\Rector\FuncCall\MysqlPConnectToMysqliConnectRector::class);
    //$rectorConfig->rule(\Rector\MysqlToMysqli\Rector\FuncCall\MysqlQueryMysqlErrorWithLinkRector::class);


    // define sets of rules
    $rectorConfig->sets([
		SetList::MYSQL_TO_MYSQLI
    ]);
};
```


Can someone explain why this is happening?

Second, per the rule for "MysqlQueryMysqlErrorWithLinkRector",
shouldn't `mysql_error();` be changed to `mysqli_error($conn);`
and
`mysql_query` be changed to `mysqli_query`
??

All files that I am scanning have many instances of these. It's happening on all files, so either I'm using some incorrect settings, or there's a bug.
Appreciate any assistance

**Reply by 7thstorm** [reply]
Tried running in phpstorm with the --debug option

`Unrecognized token 'latest': was expecting (JSON String, Number, Array, Object or token 'null', 'true' or 'false') at [Source: (String)"latest-06-169-2023/subfolder/myfile.php [file] latest-06-169-2023/subfolder/myfile.php [rule] Rector\MysqlToMysqli\Rector\Assign\MysqlAssignToMysqliRector --- consumed 0 B, total 60 MB, took 0.00 s [file] latest-06-169-2023/subfolder/myfile.php [rule] Rector\MysqlToMysqli\Rector\FuncCall\MysqlFuncCallToMysqliRector --- consumed 0 B, total 60 MB, took 0.00 s [file] latest-06-169-2023/subfolder/myfile.php [rule] Rector\MysqlToMysqli\Rector\FuncCall\MysqlPConnectToMysqliConnectRector --- consumed"[truncated 3083 chars]; line: 1, column: 7]`

Again, this happens on any file I scan against

Hope this provides enough info to help troubleshoot

**Reply by samsonasik** [reply]
We unfortunatelly can't help without reproducible code. Read this blog post, slowly

https://tomasvotruba.com/blog/2021/02/01/effective-debug-tricks-narrow-scoping/
